pkg/util/strings/matcher.go

```diff
@@ -16,11 +16,19 @@ import (
 type Matcher struct {
 	data        []string
 	matchPrefix bool
+	tags        map[string]TagMatcher
```

Comment:
Why this is here? None of the tags matching code interacts with string matching, and vice versa.

Reply:
Have moved the TagMatcher into a separate package under `pkg/filterlist`. I will be looking to pull more of the filterlist functionality out of the dogstatsd component shortly, so it is likely that would live there too.